### PR TITLE
[Report Page] Add pathogen labels

### PR DESCRIPTION
**Pull request by j-x-han**
# Description

Add pathogen labels to the new report page.
image

# Tests

* Verified that rows for  genera with pathogenic species are marked with the pathogen preview.
* Verified that rows for pathogenic species are marked with the appropriate priority label.

### CODE REVIEW

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx [hunk 1]
```diff
@@ -190,10 +191,14 @@ export default class SampleViewV2 extends React.Component {
             }
           );
         });
+        let generaPathogenCounts = getGeneraPathogenCounts(
```
Comment (jshoe) [inline app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: nit: `const generaPathogenCounts` ?

File: app/assets/src/helpers/taxon.js [hunk 2]
```diff
@@ -22,3 +22,19 @@ export const getGeneraContainingTags = taxInfoArray => {
   }
   return generaContainingTags;
 };
+
+export const getGeneraPathogenCounts = speciesCounts => {
+  let generaPathogenCounts = {};
```
Comment (jshoe) [inline app/assets/src/helpers/taxon.js]: nit: this can still be `const generaPathogenCounts`

File: app/services/pipeline_report_service.rb [hunk 3]
```diff
@@ -446,6 +449,25 @@ def find_taxa_to_highlight(sorted_genus_tax_ids, counts_by_tax_level)
     return highlighted_tax_ids
   end
 
+  def tag_pathogens(counts_by_tax_level, lineage_by_tax_id)
+    counts_by_tax_level[1].each do |species_tax_id, species_info|
```
Comment (tfrcarvalho) [inline app/services/pipeline_report_service.rb]: Use the constant value: `TaxonCount::TAX_LEVEL_SPECIES`

File: app/services/pipeline_report_service.rb [hunk 4]
```diff
@@ -446,6 +449,25 @@ def find_taxa_to_highlight(sorted_genus_tax_ids, counts_by_tax_level)
...
+      lineage = lineage_by_tax_id[species_tax_id] ? lineage_by_tax_id[species_tax_id] : lineage_by_tax_id[species_info[:genus_tax_id]]
+      name_columns = lineage.keys.select { |cn| cn.include?("_name") }
+      pathogen_tags = []
+      TaxonLineage::PRIORITY_PATHOGENS.each do |category, pathogen_list|
+        pathogen_tags |= [category] if pathogen_list.include?(species_info[:name])
+        genus_name = counts_by_tax_level[2][species_info[:genus_tax_id]][:name]
```
Comment (tfrcarvalho) [inline app/services/pipeline_report_service.rb]: Same here: `TaxonCount::TAX_LEVEL_GENUS`

File: app/assets/src/helpers/taxon.js [hunk 5]
```diff
@@ -22,3 +22,19 @@ export const getGeneraContainingTags = taxInfoArray => {
   }
   return generaContainingTags;
 };
+
+export const getGeneraPathogenCounts = speciesCounts => {
+  const generaPathogenCounts = {};
+  for (let speciesInfo of Object.values(speciesCounts)) {
```
Comment (tfrcarvalho) [inline app/assets/src/helpers/taxon.js]: Style note: We have been using `lodash/fp` like :`values(speciesCounts).forEach(`.
Up to you if you want to change.

**Pull request merged**